src/index.html

```diff
@@ -13,5 +13,151 @@
+        <div>
+          <label for="surname">Surname:</label>
+          <input 
+            type="text" 
+            id="surname" 
+            autocomplete="off"
+            required
+            pattern="\w+"
+          >
+        </div>
+        <div>
+          <label for="name">Name:</label>
+          <input type="text" id="name" autocomplete="off"
+          required
+          pattern="\w+"
+          >
+        </div>
+        <div>
+          <label for="age">How old are you?</label>
+          <input type="number" id="age"
+          required value="12"
+          min="1"
+          max="100"
+          >
+        </div>
+        <div>
+          <label for="dob">Full date of birth:</label>
+          <input 
+            type="date" 
+            id="dob" 
+            placeholder="dd/mm/yyyy"
+            required
+          >
+        </div>
+        <div>
+          <label for="terms">I accept the term of agreement</label>
+          <input 
+            type="checkbox" 
+            id="terms" 
+            required
+          >
+        </div>
+      </fieldset>
+      <fieldset>
+        <legend>Registration</legend>
+        <div>
+          <label for="email">E-mail:</label>
+          <input 
+            type="email" 
+            id="email" 
+            placeholder="email@example.com" 
+            required autocomplete="off"
+          >
+        </div>
+        <div>
+          <label for="password">Password:</label>
+          <input type="password" id="password" required
+          autocomplete="off"
+          minlength="5"
+          maxlength="15"
+          >
+        </div>
+      </fieldset>
+      <fieldset>
+        <legend>An interesting fact about you!</legend>
+        <div>
+        Do you love cats?
+          <label> <input type="radio" name="question" value="yes">Yes</label>
+  
+          <label>  <input type="radio" name="question" value="no">No</label>
+      
+        </div>
+      
+        <div>
+          <label>
+        What is your favourite color?
+        
+            <input type="color" required value="Black">
+              
+            </input>
+          </label>
+        </div>
+        <div>
+          <label>
+            What time do you go to bed?
+            <input type="time" id="time" name="time">
+          </label>
+        </div>
+        <div>
+          <label>
+            What are your favourite brand of cars
+            <select>
+              <option value="1">BMW</option>
+              <option value="2">Audi</option>
+              <option value="3">Lada</option>
+            </select>
+
+          </label>
+        </div>
+        <div>
+          <label>
+            How do you rate our work?
+            <input 
+              type="range" 
+              id="range" 
+              name="range"
+              min="0"
+              max="10"
+            >
+          </label>
+        </div>
+      </fieldset>
+      
+      
+      <fieldset>
```

Comment:
could be only one empty line btw siblings

```suggestion
      </fieldset>
      
      <fieldset>
```